common/persistence/sql/sqlplugin/sqlite/dsn.go

```diff
@@ -0,0 +1,188 @@
+// PRAGMA attributes should start with "_pragma." prefix
+// example: "_pragma.journal_mode":"wal" will be transformed to "_pragma=journal_mode(wal)"
+// More about PRAGMA attributes: https://sqlite.org/pragma.html
+// Default PRAGMA values if not provided:
+// - journal_mode: WAL (file only) 	- https://sqlite.org/pragma.html#pragma_journal_mode
+// - busy_timeout: 60 seconds 		- https://sqlite.org/pragma.html#pragma_busy_timeout
+func buildDSN(cfg *config.SQL) string {
+
+	// by default, we use in-memory database if no database name is provided
+	var dsn = "file::memory:"
+
+	// if database name is provided, then sqlite will use the file as the database
+	if cfg.DatabaseName != "" {
+		dsn = fmt.Sprintf("file:%s", cfg.DatabaseName)
+
+	}
+
+	if dsnAttrs := buildDSNAttrs(cfg); dsnAttrs != "" {
+		dsn += "?" + dsnAttrs
+	}
+
+	return dsn
+}
+
+const (
+	// if journal mode is not provided, we set it to WAL by default
+	// WAL mode allows readers and writers from different processes
+	// to access the database concurrently by default
+	// https://sqlite.org/pragma.html#pragma_journal_mode
+	// https://www.sqlite.org/wal.html
+	pragmaJournalModeAttrName     = "_pragma.journal_mode"
+	pragmaJournalModeDefaultValue = "WAL"
+
+	// if busy_timeout is not provided, we set it to 60 seconds by default
+	// https://sqlite.org/pragma.html#pragma_busy_timeout
+	pragmaBusyTimeoutAttrName = "_pragma.busy_timeout"
+	pragmaBusyTimeoutDefault  = "60000"
+)
+
+// buildDSNAttrs builds the data source name attributes for sqlite from config.SQL
+func buildDSNAttrs(cfg *config.SQL) string {
+
+	sanitizedAttrs := sanitizeDSNAttrs(cfg.ConnectAttributes)
+
+	if cfg.DatabaseName != "" {
+		defaultIfEmpty(sanitizedAttrs, pragmaJournalModeAttrName, pragmaJournalModeDefaultValue)
+	}
+
+	defaultIfEmpty(sanitizedAttrs, pragmaBusyTimeoutAttrName, pragmaBusyTimeoutDefault)
+	return joinDSNAttrs(sanitizedAttrs)
+}
+
+// defaultIfEmpty sets the value to the key if the key is not present in the attributes
+func defaultIfEmpty(attrs map[string]string, key, value string) {
+	if hasAttr(attrs, key) {
+		return
+	}
+	attrs[key] = value
+}
+
+// hasAttr checks if the attributes map has any of the keys
+func hasAttr(attrs map[string]string, keys ...string) bool {
+	for key := range attrs {
+		for _, k := range keys {
+			if key == k {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+// sanitizeDSNAttrs sanitizes the attributes by trimming the keys and values
+func sanitizeDSNAttrs(attrs map[string]string) map[string]string {
+	sanitized := make(map[string]string, len(attrs))
+
+	for k, v := range attrs {
+		k, v = sanitizeDSNAttrKey(k), sanitizeDSNAttrValue(v)
+		sanitized[k] = v
+	}
+
+	return sanitized
+}
+
+const (
+	// pragmaKey is the key is used to set pragma arguments in the DSN
+	pragmaKey = "_pragma"
+
+	// pragmaPrefix is the prefix used to identify pragma arguments in the config
+	pragmaPrefix = "_pragma."
+)
```

Comment:
I suggest having all the consts in the top of the file

Reply:
thanks, moved them 👍